lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -41,11 +41,11 @@ def initialize(cmd)
         at_exit { exit }
       end
 
-      def execute(powershell_code, timeout_ms = 300 * 1000)
+      def execute(powershell_code, timeout_ms = nil, working_dir = nil)
```

Comment:
1. `timeout_ms` ultimately gets used in the `make_ps_code` function where it sets the default for 300*1000.
2. resource[:timeout] incidentally is set to `300.0` if you don't specify anything in the manifest
3. This is protection for when the method is called directly (mainly tests).  Normally this is only called via the provider